omod/src/main/webapp/fragments/field/personRelationship.gsp

```diff
@@ -16,12 +16,22 @@
             <select id="relationship_type" name="relationship_type" class="rel_type" ng-model="relationship.type">
                 <option value="">${ui.message('registrationapp.person.relationship.selectRelationshipType')}</option>
                 <% relationshipTypes.each { type -> %>
-                <option value="${type.uuid}-A">${ui.encodeHtmlContent(ui.message(type.aIsToB))}</option>
+                    <option value="${type.uuid}-A" data-val="${type.aIsToB}">
+                        <% def msgString = "ui.i18n.RelationshipType.aIsToB." + type.uuid %>
+                        ${ui.message(msgString) == msgString ?
+                                type.aIsToB :
```

Comment:
Shouldn't the line above be? `ui.encodeHtmlContent(ui.message(type.aIsToB))`

Reply:
I don't think so, because the intention is to check whether `ui.message` just returned the input message code. The idea is to display the relationship name as a fallback instead of rendering the code itself. It's surprising to me that `ui.message` doesn't have an option to accept a fallback, but anyway, that's what's going on here.

Reply:
fwiw, @brandones  you may have already seen this the idea here is that "ui.format(type)" should probably handle it, but it looks like we don't have a handler for RelationshipType

For reference, here's the underlying method that ui.format will call:

https://github.com/openmrs/openmrs-module-uiframework/blob/master/api/src/main/java/org/openmrs/ui/framework/UiUtils.java#L434

Which ends up calling this methods:

https://github.com/openmrs/openmrs-module-uiframework/blob/master/api/src/main/java/org/openmrs/ui/framework/FormatterImpl.java#L63

And most of these end up calling this:

https://github.com/openmrs/openmrs-module-uiframework/blob/master/api/src/main/java/org/openmrs/ui/framework/FormatterImpl.java#L152

This would only support a single localized display text per RelationshipType, not an aIsToB and vice versa...

Anyway, long story short, this looks good, but we may want to consider building out handling for this in ui.format in the future.